modules/mcagar/src/main/ts/ephox/mcagar/api/Editor.ts

```diff
@@ -29,6 +31,14 @@ const cFromElement = <T extends EditorType = EditorType>(element: SugarElement,
 
     const targetSettings = SugarShadowDom.isInShadowRoot(element) ? ({ target: element.dom }) : ({ selector: '#' + randomId });
 
+    const onRemove = (e: { editor: EditorType }) => {
+      if (e.editor.id === randomId) {
+        tinymce.off('RemoveEditor', onRemove);
+        Selectors.one('#' + randomId).each(Remove.remove);
+        die(errorMessageEditorRemoved);
+      }
+    };
+    tinymce.on('RemoveEditor', onRemove);
```

Comment:
Is there a reason you didn't put this in the editor setup callback? That seems like it'd be a better solution here, as you wouldn't need to compare the ids. I'm guessing it was done to ensure there's absolutely no way to have the remove call be fired before setup ran.

Note: Not asking for anything to be changed here, I'm just curious.

Reply:
Basically because `editor.on('remove' ...)` is not documented and I didn't realise it existed.

Reply:
Nevermind, it is documented but I didn't find it when I looked because I was looking for something called unload.

Reply:
I switched it to use the editor 'remove' event.